packages/at_repl/bin/at_repl.dart

```diff
@@ -1,53 +1,77 @@
 import 'dart:async';
+import 'dart:io';
 import 'package:at_repl/repl.dart';
 import 'package:at_utils/at_utils.dart';
 import 'package:io/ansi.dart';
 import 'package:args/args.dart';
-Future<void> main(List<String> args) async {
 
-  // ArgParser, look for -a <atSign>, and --rootUrl <rootUrl> defaults to root.atsign.org:64, and --verbose or -v for verbose output
+Future<void> main(List<String> args) async {
+  // ArgParser, look for -a <atSign>, and --root-domain (<host>[:port]) defaults to root.atsign.org:64, and --verbose or -v for verbose output
   final parser = ArgParser()
-    ..addOption('atSign', abbr: 'a', help: 'The atSign to use')
-    ..addOption('rootUrl', defaultsTo: 'root.atsign.org:64', help: 'The root URL to connect to')
-    ..addOption('keys-file', abbr: 'k', help: 'Path to the atKeys file')
-    ..addFlag('verbose', abbr: 'v', defaultsTo: false, help: 'Enable verbose output')
-    ..addFlag('help', abbr: 'h', defaultsTo: false, help: 'Show this help message');
+    ..addOption('atSign',
+      abbr: 'a',
+      help: 'The atSign to use',
+      mandatory: true)
+    ..addOption('root-domain',
+      defaultsTo: 'root.atsign.org:64',
+      help: 'The root domain (optionally host:port) to connect to',
+      aliases: ['rootUrl'])
+    ..addOption('key-file',
+      help: 'Path to the .atKeys file',
+      abbr: 'k')
+    ..addFlag('verbose',
+      abbr: 'v',
+      defaultsTo: false,
+      help: 'Enable verbose output')
+    ..addFlag('help',
+      abbr: 'h',
+      defaultsTo: false,
+      help: 'Show this help message')
+    ..addFlag('version',
+      defaultsTo: false,
+      help: 'Show at_repl version');
+
+  if (args.contains('--version') || args.contains('-V')) {
+    _printVersion();
+    return;
+  }
+
+  if (args.contains('-h') || args.contains('--help')) {
+    _printUsage(parser);
+    return;
+  }
 
   late ArgResults results;
   try {
     results = parser.parse(args);
   } catch (e) {
     print('Error: $e');
     print('');
-    print('Usage: at_repl [options]');
-    print('');
-    print('Options:');
-    print(parser.usage);
-    return;
-  }
-
-  // Check if help flag is set
-  if (results['help'] as bool) {
-    print('Usage: at_repl [options]');
-    print('');
-    print('Options:');
-    print(parser.usage);
+    _printUsage(parser);
     return;
   }
 
   // Check if required atSign is provided
-  final String? atSign = results['atSign'] as String?;
-  if (atSign == null || atSign.isEmpty) {
+  final String? atSignArg = results['atSign'] as String?;
+  if (atSignArg == null || atSignArg.trim().isEmpty) {
     print('Error: atSign is required');
```

Comment:
it's always required because you are setting `mandatory:true` when adding the option